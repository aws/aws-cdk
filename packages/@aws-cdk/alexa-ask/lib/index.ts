// Alexa::ASK CloudFormation Resources:
export * from './ask.generated';
export * from './pipeline-actions';
