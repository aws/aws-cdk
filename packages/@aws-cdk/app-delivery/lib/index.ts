export * from './pipeline-deploy-stack-action';
