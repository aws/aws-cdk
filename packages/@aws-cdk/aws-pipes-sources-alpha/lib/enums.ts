/**
 * Define how to handle item process failures.
 */
export enum OnPartialBatchItemFailure {
  /**
   * EventBridge halves each batch and will retry each half until all the
   * records are processed or there is one failed message left in the batch.
   */
  AUTOMATIC_BISECT = 'AUTOMATIC_BISECT',
}

/**
 * The position in a Kinesis stream from which to start reading.
 */
export enum KinesisStartingPosition {
  /**
   * Start streaming at the last untrimmed record in the shard,
   * which is the oldest data record in the shard.
   */
  TRIM_HORIZON = 'TRIM_HORIZON',
  /**
   * Start streaming just after the most recent record in the shard,
   * so that you always read the most recent data in the shard.
   */
  LATEST = 'LATEST',
  /**
   * Start streaming from the position denoted by the time stamp
   * specified in the `startingPositionTimestamp` field.
   */
  AT_TIMESTAMP = 'AT_TIMESTAMP',
}

/**
 * The position in a DynamoDB stream from which to start reading.
 */
export enum DynamoDBStartingPosition {
  /**
   * Start reading at the last (untrimmed) stream record,
   * which is the oldest record in the shard.
   */
  TRIM_HORIZON = 'TRIM_HORIZON',
  /**
   * Start reading just after the most recent stream record in the shard,
   * so that you always read the most recent data in the shard.
   */
  LATEST = 'LATEST',
}
