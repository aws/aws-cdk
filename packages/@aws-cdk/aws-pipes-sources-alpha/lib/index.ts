export * from './dynamodb';
export * from './enums';
export * from './kinesis';
export * from './sqs';
export * from './streamSource';
