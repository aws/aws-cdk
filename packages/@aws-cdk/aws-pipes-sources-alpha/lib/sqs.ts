import { IPipe, ISource, SourceConfig } from '@aws-cdk/aws-pipes-alpha';
import { Duration } from 'aws-cdk-lib';
import { IRole } from 'aws-cdk-lib/aws-iam';
import { IQueue } from 'aws-cdk-lib/aws-sqs';

/**
 * Parameters for the SQS source.
 */
export interface SqsSourceParameters {
  /**
    * The maximum number of records to include in each batch.
    *
    * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcesqsqueueparameters.html#cfn-pipes-pipe-pipesourcesqsqueueparameters-batchsize
    * @default 10
    */
  readonly batchSize?: number;

  /**
   * The maximum length of a time to wait for events.
   *
   * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcesqsqueueparameters.html#cfn-pipes-pipe-pipesourcesqsqueueparameters-maximumbatchingwindowinseconds
   * @default 1
   */
  readonly maximumBatchingWindow?: Duration;
}

/**
 * A source that reads from an SQS queue.
 */
export class SqsSource implements ISource {
  private readonly queue: IQueue;
  readonly sourceArn;
  private sourceParameters;

  constructor(queue: IQueue, parameters?: SqsSourceParameters) {
    this.queue = queue;
    this.sourceArn = queue.queueArn;
    if (parameters) {
      this.sourceParameters = parameters;
    }
  }

  bind(_pipe: IPipe): SourceConfig {
    if (!this.sourceParameters) {
      return {};
    }

    return {
      sourceParameters: {
        sqsQueueParameters: {
          batchSize: this.sourceParameters?.batchSize,
          maximumBatchingWindowInSeconds: this.sourceParameters?.maximumBatchingWindow?.toSeconds(),
        },
      },
    };
  }

  grantRead(grantee: IRole): void {
    this.queue.grantConsumeMessages(grantee);
  }

}

