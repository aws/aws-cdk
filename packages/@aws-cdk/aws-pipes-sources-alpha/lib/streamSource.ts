import { SourceWithDeadLetterTarget } from '@aws-cdk/aws-pipes-alpha';
import { Duration, Token } from 'aws-cdk-lib';
import { ITopic } from 'aws-cdk-lib/aws-sns';
import { IQueue } from 'aws-cdk-lib/aws-sqs';
import { OnPartialBatchItemFailure } from './enums';

/**
 * Base parameters for streaming sources.
 */
export interface StreamSourceParameters {
  /**
   * The maximum number of records to include in each batch.
   *
   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-batchsize
   * @default 1
   */
  readonly batchSize?: number;

  /**
   * Define the target to send dead-letter queue events to.
   *
   * The dead-letter queue stores any events that are not successfully delivered to a Pipes target after all retry attempts are exhausted.
   * You can then resolve the issue that caused the failed invocations and replay the events at a later time.
   * In some cases, such as when access is denied to a resource, events are sent directly to the dead-letter queue and are not retried.
   *
   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-deadletterconfig
   * @default - no dead-letter queue or topic
   */
  readonly deadLetterTarget?: IQueue | ITopic;

  /**
   * The maximum length of a time to wait for events.
   *
   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-maximumbatchingwindowinseconds
   * @default - the events will be handled immediately
   */
  readonly maximumBatchingWindow?: Duration;

  /**
   * Discard records older than the specified age. The default value is -1, which sets the maximum age to infinite. When the value is set to infinite, EventBridge never discards old records.
   *
   *  @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-maximumrecordageinseconds
   * @default -1 - EventBridge won't discard old records
   */
  readonly maximumRecordAge?: Duration;

  /**
   * Discard records after the specified number of retries. The default value is -1, which sets the maximum number of retries to infinite. When MaximumRetryAttempts is infinite, EventBridge retries failed records until the record expires in the event source.
   *
   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-maximumretryattempts
   * @default -1 - EventBridge will retry failed records until the record expires in the event source
   */
  readonly maximumRetryAttempts?: number;

  /**
   * Define how to handle item process failures. {@link OnPartialBatchItemFailure.AUTOMATIC_BISECT} halves each batch and will retry each half until all the records are processed or there is one failed message left in the batch.
   *
   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-onpartialbatchitemfailure
   * @default off - EventBridge will retry the entire batch
   */
  readonly onPartialBatchItemFailure?: OnPartialBatchItemFailure;

  /**
   * The number of batches to process concurrently from each shard.
   *
   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-parallelizationfactor
   * @default 1
   */
  readonly parallelizationFactor?: number;
}

/**
 * Streaming sources.
 */
export abstract class StreamSource extends SourceWithDeadLetterTarget {
  /**
   * The ARN of the source resource.
   */
  readonly sourceArn: string;
  /**
   * Base parameters for streaming sources.
   */
  readonly sourceParameters: StreamSourceParameters;

  constructor(sourceArn: string, sourceParameters: StreamSourceParameters) {
    super(sourceArn, sourceParameters.deadLetterTarget);
    this.sourceArn = sourceArn;
    this.sourceParameters = sourceParameters;

    validateBatchSize(this.sourceParameters.batchSize);
    validateMaximumBatchingWindow(this.sourceParameters.maximumBatchingWindow?.toSeconds());
    validateMaximumRecordAge(this.sourceParameters.maximumRecordAge?.toSeconds());
    validateMaxiumRetryAttemps(this.sourceParameters.maximumRetryAttempts);
    validateParallelizationFactor(this.sourceParameters.parallelizationFactor);
  }
}

function validateBatchSize(batchSize?: number) {
  if (batchSize !== undefined && !Token.isUnresolved(batchSize)) {
    if (batchSize < 1 || batchSize > 10000) {
      throw new Error(`Batch size must be between 1 and 10000, received ${batchSize}`);
    }
  }
}

function validateMaximumBatchingWindow(window?: number) {
  if (window !== undefined) {
    // only need to check upper bound since Duration amounts cannot be negative
    if (window > 300) {
      throw new Error(`Maximum batching window must be between 0 and 300, received ${window}`);
    }
  }
}

function validateMaximumRecordAge(age?: number) {
  if (age !== undefined) {
    if (age < 60 || age > 604800) {
      throw new Error(`Maximum record age in seconds must be between 60 and 604800 (leave undefined for infinite), received ${age}`);
    }
  }
}

function validateMaxiumRetryAttemps(attempts?: number) {
  if (attempts !== undefined && !Token.isUnresolved(attempts)) {
    if (attempts < -1 || attempts > 10000) {
      throw new Error(`Maximum retry attempts must be between -1 and 10000, received ${attempts}`);
    }
  }
}

function validateParallelizationFactor(factor?: number) {
  if (factor !== undefined && !Token.isUnresolved(factor)) {
    if (factor < 1 || factor > 10) {
      throw new Error(`Parallelization factor must be between 1 and 10, received ${factor}`);
    }
  }
}
