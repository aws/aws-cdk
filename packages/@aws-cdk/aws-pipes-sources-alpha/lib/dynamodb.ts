import { IPipe, SourceConfig } from '@aws-cdk/aws-pipes-alpha';
import { ITableV2 } from 'aws-cdk-lib/aws-dynamodb';
import { IRole } from 'aws-cdk-lib/aws-iam';
import { DynamoDBStartingPosition } from './enums';
import { StreamSource, StreamSourceParameters } from './streamSource';

/**
 * Parameters for the DynamoDB source.
 */
export interface DynamoDBSourceParameters extends StreamSourceParameters {
  /**
   * The position in a stream from which to start reading.
   *
   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcedynamodbstreamparameters.html#cfn-pipes-pipe-pipesourcedynamodbstreamparameters-startingposition
   */
  readonly startingPosition: DynamoDBStartingPosition;
}

/**
 * A source that reads from an DynamoDB stream.
 */
export class DynamoDBSource extends StreamSource {
  private readonly table: ITableV2;
  private readonly startingPosition: DynamoDBStartingPosition;
  private readonly deadLetterTargetArn?: string;

  constructor(table: ITableV2, parameters: DynamoDBSourceParameters) {
    if (table.tableStreamArn === undefined) {
      throw new Error('Table does not have a stream defined, cannot create pipes source');
    }

    super(table.tableStreamArn, parameters);
    this.table = table;
    this.startingPosition = parameters.startingPosition;
    this.deadLetterTargetArn = this.getDeadLetterTargetArn(this.deadLetterTarget);
  }

  bind(_pipe: IPipe): SourceConfig {
    return {
      sourceParameters: {
        dynamoDbStreamParameters: {
          batchSize: this.sourceParameters.batchSize,
          deadLetterConfig: this.deadLetterTargetArn ? { arn: this.deadLetterTargetArn } : undefined,
          maximumBatchingWindowInSeconds: this.sourceParameters.maximumBatchingWindow?.toSeconds(),
          maximumRecordAgeInSeconds: this.sourceParameters.maximumRecordAge?.toSeconds(),
          maximumRetryAttempts: this.sourceParameters.maximumRetryAttempts,
          onPartialBatchItemFailure: this.sourceParameters.onPartialBatchItemFailure,
          parallelizationFactor: this.sourceParameters.parallelizationFactor,
          startingPosition: this.startingPosition,
        },
      },
    };
  }

  grantRead(grantee: IRole): void {
    this.table.grantStreamRead(grantee);
  }
}
