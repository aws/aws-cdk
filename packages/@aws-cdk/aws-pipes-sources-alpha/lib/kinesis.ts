import { IPipe, SourceConfig } from '@aws-cdk/aws-pipes-alpha';
import { IRole } from 'aws-cdk-lib/aws-iam';
import { IStream } from 'aws-cdk-lib/aws-kinesis';
import { KinesisStartingPosition } from './enums';
import { StreamSource, StreamSourceParameters } from './streamSource';

/**
 * Parameters for the Kinesis source.
 */
export interface KinesisSourceParameters extends StreamSourceParameters {
  /**
   * The position in a stream from which to start reading.
   *
   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-startingposition
   */
  readonly startingPosition: KinesisStartingPosition;

  /**
   * With StartingPosition set to AT_TIMESTAMP, the time from which to start reading, in ISO 8601 format.
   *
   * @example
   * new Date(Date.UTC(1969, 10, 20, 0, 0, 0))
   *
   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-startingpositiontimestamp
   * @default - no starting position timestamp
   */
  readonly startingPositionTimestamp?: Date;
}

/**
 * A source that reads from Kinesis.
 */
export class KinesisSource extends StreamSource {
  private readonly stream: IStream;
  private readonly startingPosition: KinesisStartingPosition;
  private readonly startingPositionTimestamp?: Date;
  private readonly deadLetterTargetArn?: string;

  constructor(stream: IStream, parameters: KinesisSourceParameters) {
    super(stream.streamArn, parameters);
    this.stream = stream;
    this.startingPosition = parameters.startingPosition;
    this.startingPositionTimestamp = parameters.startingPositionTimestamp;
    this.deadLetterTargetArn = this.getDeadLetterTargetArn(this.deadLetterTarget);

    if (this.startingPositionTimestamp && this.startingPosition !== KinesisStartingPosition.AT_TIMESTAMP) {
      throw new Error(`Timestamp only valid with StartingPosition AT_TIMESTAMP for Kinesis streams, received ${this.startingPosition}`);
    }
  }

  bind(_pipe: IPipe): SourceConfig {
    return {
      sourceParameters: {
        kinesisStreamParameters: {
          batchSize: this.sourceParameters.batchSize,
          deadLetterConfig: this.deadLetterTargetArn ? { arn: this.deadLetterTargetArn } : undefined,
          maximumBatchingWindowInSeconds: this.sourceParameters.maximumBatchingWindow?.toSeconds(),
          maximumRecordAgeInSeconds: this.sourceParameters.maximumRecordAge?.toSeconds(),
          maximumRetryAttempts: this.sourceParameters.maximumRetryAttempts,
          onPartialBatchItemFailure: this.sourceParameters.onPartialBatchItemFailure,
          parallelizationFactor: this.sourceParameters.parallelizationFactor,
          startingPosition: this.startingPosition,
          startingPositionTimestamp: this.startingPositionTimestamp?.toISOString(),
        },
      },
    };
  }

  grantRead(grantee: IRole): void {
    this.stream.grantRead(grantee);
  }
}
