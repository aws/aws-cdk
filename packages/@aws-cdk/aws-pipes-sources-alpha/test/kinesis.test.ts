import { Pipe } from '@aws-cdk/aws-pipes-alpha';
import { App, Duration, Stack } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { Stream } from 'aws-cdk-lib/aws-kinesis';
import { Topic } from 'aws-cdk-lib/aws-sns';
import { Queue } from 'aws-cdk-lib/aws-sqs';
import { TestTarget } from './test-classes';
import { KinesisSource, KinesisStartingPosition, OnPartialBatchItemFailure } from '../lib';

describe('kinesis source', () => {
  it('should have only source arn and starting position', () => {
    // ARRANGE
    const app = new App();
    const stack = new Stack(app, 'TestStack');
    const stream = new Stream(stack, 'MyStream', {});
    const source = new KinesisSource(stream, {
      startingPosition: KinesisStartingPosition.TRIM_HORIZON,
    });

    new Pipe(stack, 'MyPipe', {
      source,
      target: new TestTarget(),
    });

    // ACT
    const template = Template.fromStack(stack);

    // ASSERT
    template.hasResourceProperties('AWS::Pipes::Pipe', {
      Source: {
        'Fn::GetAtt': [
          'MyStream5C050E93',
          'Arn',
        ],
      },
      SourceParameters: {
        KinesisStreamParameters: {
          StartingPosition: 'TRIM_HORIZON',
        },
      },
    });
  });

  it('should have source parameters (dlq = sns)', () => {
    // ARRANGE
    const app = new App();
    const stack = new Stack(app, 'TestStack');
    const stream = new Stream(stack, 'MyStream', {});
    const topic = new Topic(stack, 'MyTopic', {});
    const source = new KinesisSource(stream, {
      batchSize: 10,
      deadLetterTarget: topic,
      maximumBatchingWindow: Duration.seconds(10),
      maximumRecordAge: Duration.seconds(60),
      maximumRetryAttempts: 10,
      onPartialBatchItemFailure: OnPartialBatchItemFailure.AUTOMATIC_BISECT,
      parallelizationFactor: 10,
      startingPosition: KinesisStartingPosition.AT_TIMESTAMP,
      startingPositionTimestamp: new Date(Date.UTC(2024, 0, 1, 0, 0, 0)),
    });

    new Pipe(stack, 'MyPipe', {
      source,
      target: new TestTarget(),
    });

    // ACT
    const template = Template.fromStack(stack);

    // ASSERT
    template.hasResourceProperties('AWS::Pipes::Pipe', {
      Source: {
        'Fn::GetAtt': [
          'MyStream5C050E93',
          'Arn',
        ],
      },
      SourceParameters: {
        KinesisStreamParameters: {
          BatchSize: 10,
          DeadLetterConfig: {
            Arn: { Ref: 'MyTopic86869434' },
          },
          MaximumBatchingWindowInSeconds: 10,
          MaximumRecordAgeInSeconds: 60,
          MaximumRetryAttempts: 10,
          OnPartialBatchItemFailure: 'AUTOMATIC_BISECT',
          ParallelizationFactor: 10,
          StartingPosition: 'AT_TIMESTAMP',
          StartingPositionTimestamp: '2024-01-01T00:00:00.000Z',
        },
      },
    });
  });

  it('should have source parameters (dlq = sqs)', () => {
    // ARRANGE
    const app = new App();
    const stack = new Stack(app, 'TestStack');
    const stream = new Stream(stack, 'MyStream', {});
    const queue = new Queue(stack, 'MyQueue', {});
    const source = new KinesisSource(stream, {
      batchSize: 10,
      deadLetterTarget: queue,
      maximumBatchingWindow: Duration.seconds(10),
      maximumRecordAge: Duration.seconds(60),
      maximumRetryAttempts: 10,
      onPartialBatchItemFailure: OnPartialBatchItemFailure.AUTOMATIC_BISECT,
      parallelizationFactor: 10,
      startingPosition: KinesisStartingPosition.AT_TIMESTAMP,
      startingPositionTimestamp: new Date(Date.UTC(2024, 0, 1, 0, 0, 0)),
    });

    new Pipe(stack, 'MyPipe', {
      source,
      target: new TestTarget(),
    });

    // ACT
    const template = Template.fromStack(stack);

    // ASSERT
    template.hasResourceProperties('AWS::Pipes::Pipe', {
      Source: {
        'Fn::GetAtt': [
          'MyStream5C050E93',
          'Arn',
        ],
      },
      SourceParameters: {
        KinesisStreamParameters: {
          BatchSize: 10,
          DeadLetterConfig: {
            Arn: {
              'Fn::GetAtt': [
                'MyQueueE6CA6235',
                'Arn',
              ],
            },
          },
          MaximumBatchingWindowInSeconds: 10,
          MaximumRecordAgeInSeconds: 60,
          MaximumRetryAttempts: 10,
          OnPartialBatchItemFailure: 'AUTOMATIC_BISECT',
          ParallelizationFactor: 10,
          StartingPosition: 'AT_TIMESTAMP',
          StartingPositionTimestamp: '2024-01-01T00:00:00.000Z',
        },
      },
    });
  });

  it('should grant pipe role read access', () => {
    // ARRANGE
    const app = new App();
    const stack = new Stack(app, 'TestStack');
    const stream = new Stream(stack, 'MyStream', {});
    const source = new KinesisSource(stream, {
      startingPosition: KinesisStartingPosition.LATEST,
    });

    new Pipe(stack, 'MyPipe', {
      source,
      target: new TestTarget(),
    });

    // ACT
    const template = Template.fromStack(stack);

    // ASSERT
    expect(template.findResources('AWS::IAM::Role')).toMatchSnapshot();
    expect(template.findResources('AWS::IAM::Policy')).toMatchSnapshot();
  });
});

describe('kinesis source parameters validation', () => {
  test('timestamp provided and starting position not set to AT_TIMESTAMP should throw', () => {
    // GIVEN
    const app = new App();
    const stack = new Stack(app, 'demo-stack');
    const stream = new Stream(stack, 'MyStream', {});

    // WHEN
    expect(() => {
      new KinesisSource(stream, {
        startingPosition: KinesisStartingPosition.LATEST,
        startingPositionTimestamp: new Date(Date.UTC(2024, 0, 1, 0, 0, 0)),
      });
    }).toThrow('Timestamp only valid with StartingPosition AT_TIMESTAMP for Kinesis streams, received LATEST');
  });
});
