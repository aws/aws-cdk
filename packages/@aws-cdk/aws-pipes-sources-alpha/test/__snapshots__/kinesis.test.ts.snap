// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kinesis source should grant pipe role read access 1`] = `
{
  "MyPipeRoleCBC8E9AB": {
    "Properties": {
      "AssumeRolePolicyDocument": {
        "Statement": [
          {
            "Action": "sts:AssumeRole",
            "Effect": "Allow",
            "Principal": {
              "Service": "pipes.amazonaws.com",
            },
          },
        ],
        "Version": "2012-10-17",
      },
    },
    "Type": "AWS::IAM::Role",
  },
}
`;

exports[`kinesis source should grant pipe role read access 2`] = `
{
  "MyPipeRoleDefaultPolicy31387C20": {
    "Properties": {
      "PolicyDocument": {
        "Statement": [
          {
            "Action": [
              "kinesis:DescribeStreamSummary",
              "kinesis:GetRecords",
              "kinesis:GetShardIterator",
              "kinesis:ListShards",
              "kinesis:SubscribeToShard",
              "kinesis:DescribeStream",
              "kinesis:ListStreams",
              "kinesis:DescribeStreamConsumer",
            ],
            "Effect": "Allow",
            "Resource": {
              "Fn::GetAtt": [
                "MyStream5C050E93",
                "Arn",
              ],
            },
          },
        ],
        "Version": "2012-10-17",
      },
      "PolicyName": "MyPipeRoleDefaultPolicy31387C20",
      "Roles": [
        {
          "Ref": "MyPipeRoleCBC8E9AB",
        },
      ],
    },
    "Type": "AWS::IAM::Policy",
  },
}
`;
